### PR TITLE
[angular-material]: IThemingProvider and IProgressCircularProvide

### PR DESCRIPTION
**Pull request by Dona278**
* Added methods "generateThemesOnDemand", "disableTheming", and
"configuration" to the IThemingProvider
* Added IProgressCircularProvider

### CODE REVIEW

**Comment by andy-ms** [comment]
Rebase against `master` to fix tests.

File: angular-material/angular-material-tests.ts [hunk 1]
```diff
@@ -4,7 +4,8 @@ var myApp = angular.module('testModule', ['ngMaterial']);
 
 myApp.config((
     $mdThemingProvider: ng.material.IThemingProvider,
-    $mdIconProvider: ng.material.IIconProvider) => {
+    $mdIconProvider: ng.material.IIconProvider,
+	$mdProgressCircularProvider: ng.material.IProgressCircularProvider) => {
```
Comment (andy-ms) [inline angular-material/angular-material-tests.ts]: This line uses a tab, but the rest of the file uses spaces.

Comment (Dona278) [inline angular-material/angular-material-tests.ts]: Fixed. However i have tried to rebase this pull request with `master` and i have got some merge conflicts. Is the first rebase for me :)

Comment (andy-ms) [inline angular-material/angular-material-tests.ts]: I think what you want is `git rebase --onto master types-2.0`. See [this question](http://stackoverflow.com/questions/40524681/why-git-rebase-shows-conflicts-in-the-files-i-did-not-modify).

Comment (Dona278) [inline angular-material/angular-material-tests.ts]: Sorry @andy-ms , i used `git rebase --onto master angular-material` but now i see other commits (master branch?) in my pull request and it's not clear. Where did I go wrong?

Comment (andy-ms) [inline angular-material/angular-material-tests.ts]: The pull request is still targeted to `types-2.0`. To target `master` you have to click "Edit" in the upper-right and change the base branch to `master`.
But this pull request no longer seems to have any commits by you! Hope you can recover your commits!

**Comment by Dona278** [comment]
@andy-ms i'm sorry, this is the remake #13724

**Pull request closed without merging**